Makefile

```diff
@@ -205,6 +205,7 @@ coverage: coverage-frontend coverage-runtime
 
 coverage-frontend:
 	@echo "Generating coverage report for the frontend"
+	$(ASAN_COMMAND) $(PYTHON) -m pip install scipy
```

Comment:
Here (and other places) as well I would be hesitant to install `scipy` manually. There should be a clear strategy about how we are bringing in the `scipy` dependency.

If a **built-time dependency**, it can be added to the relevant build scripts in the corresponding setup step.

If a **run-time dependency**, it should probably only be placed in one location: the `setup.py` package requirements. However, if the dependency is brought in by lightning rather than Catalyst directly, you would think the lightning package requirements already contain a `scipy` entry.